### PR TITLE
openvpn LDAP error handling enhancements

### PR DESCRIPTION
**Pull request by stwilliamsWFS**
These additions allow for handling of LDAP authentication error messages generated by the openvpn server when the server has been configured to off-load authentication to a LDAP server (e.g. Active Directory)

Stu Williams (Workforce Software)

### CODE REVIEW

File: rules/0400-openvpn_rules.xml [hunk 1]
```diff
@@ -50,4 +50,28 @@
...
+ Author: Stuart Williams <stwilliams@workforcesoftware.com>
+    Date: 2019-01-29
+    Note: Enhanced decoder rules to handle LDAP integrated authentication
+Comments: Not sure how useful the first entry is but will 'log' it - could be useful in auditing numbers etc
+
+Jan 28 14:25:49 VPN-SERVER-05892 openvpn: LDAP bind failed: Invalid credentials (80090308: LdapErr: DSID-55555555, comment: AcceptSecurityContext error, data 775, v3839)
+Jan 28 14:25:49 VPN-SERVER-05892 openvpn: Incorrect password supplied for LDAP DN "CN=Harry T. Hacker,OU=business unit,OU=department,DC=domain,DC=com".
+-->
+
+  <rule id="81810" level="5">
```
Comment (chemamartinez) [inline rules/0400-openvpn_rules.xml:63]: Can you change the rule ID to continue the numeric sequence of the OpenVPN rules? The last one is `81804` so this rule should be the `81805` and the other one the `81806`

File: rules/0400-openvpn_rules.xml [hunk 2]
```diff
@@ -50,4 +50,28 @@
...
+    <if_sid>81800</if_sid>
+    <match>^LDAP bind failed:</match>
+    <description>OpenVPN LDAP Bind Failure: $(ldap_data.error_message)</description>
+    <group>pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7</group>
+  </rule>
+
+  <rule id="81820" level="5">
```
Comment (chemamartinez) [inline rules/0400-openvpn_rules.xml:70]: Same as above, change the ID to 81806.

File: tools/rules-testing/tests/openvpn_ldap.ini [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[openvpn: LDAP Bind Failed]
+log 1 fail = Jan 28 14:25:49 VPN-SERVER-05892 openvpn: LDAP bind failed: Invalid credentials (80090308: LdapErr: DSID-55555555, comment: AcceptSecurityContext error, data 775, v3839)
+rule = 81810
```
Comment (chemamartinez) [inline tools/rules-testing/tests/openvpn_ldap.ini:3]: Change the IDs in this file as well

**Review by chemamartinez — CHANGES_REQUESTED** [review]
Hi @stwilliamsWFS,

Sorry for the delayed review.

Rules and decoders added look good to us and work as expected. We just have submitted a change request related to the rules IDs. As soon as the changes are made the pull request will be approved and merged.

Thank you for your contribution to the project. It is really appreciated.

Best regards,
Chema.

File: rules/0400-openvpn_rules.xml (hunk 1 above)
Comment (stwilliamsWFS) [reply rules/0400-openvpn_rules.xml]: Updated as requested.

File: rules/0400-openvpn_rules.xml (hunk 2 above)
Comment (stwilliamsWFS) [reply rules/0400-openvpn_rules.xml]: Updated as requested.

File: tools/rules-testing/tests/openvpn_ldap.ini (hunk 3 above)
Comment (stwilliamsWFS) [reply tools/rules-testing/tests/openvpn_ldap.ini]: Updated as requested.

**Review by chemamartinez — APPROVED** [review]

**Comment by chemamartinez** [comment]
Merged!

Thank you again for your contribution @stwilliamsWFS!

Best regards.

**Pull request merged**